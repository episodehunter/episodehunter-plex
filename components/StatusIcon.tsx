import * as React from 'react';
import { StatusType } from '../types';

const style = status => ({
  borderRadius: '10px',
  width: '20px',
  height: '20px',
  alignSelf: 'center',
  marginRight: '10px',
  backgroundColor: ['green', '#d2d200', 'red'][status]
});

export function StatusIcon({status}: {status: StatusType}) {
  return (
    <div style={style(status)} />
  );
}
