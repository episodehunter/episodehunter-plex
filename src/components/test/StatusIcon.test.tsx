import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { StatusIcon } from '../StatusIcon';

test('that StatusIcon renders', () => {
  const tree = renderer.create(
    <StatusIcon status={0} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
