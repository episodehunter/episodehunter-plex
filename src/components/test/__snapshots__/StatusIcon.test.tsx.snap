// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`that StatusIcon renders 1`] = `
<div
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "green",
      "borderRadius": "10px",
      "height": "20px",
      "marginRight": "10px",
      "width": "20px",
    }
  }
/>
`;
